frontend/components/Map/Map.js

```diff
@@ -1,59 +1,62 @@
 import React from 'react';
-import { Text } from 'react-native';
-
-import MapView, { PROVIDER_GOOGLE, Marker, Callout } from 'react-native-maps';
+import MapView, { PROVIDER_GOOGLE, Marker, Circle } from 'react-native-maps';
 import { connect } from 'react-redux';
 
 import { MapStyles } from './Map.Styles';
 
 class Map extends React.Component {
-    constructor() {
-        super();
+    constructor(props) {
+        super(props);
         this.state = {
+            region: {
+                latitude: props.location.latitude,
+                longitude: props.location.longitude,
+                latitudeDelta: 2,
+                longitudeDelta: 1,
+            },
             marker: [],
+            currentPositionMarker: null,
         };
     }
 
     updateMapMarkerFromLocations = (d) => {
         let data = d || [];
-      
+
         const dataMarker = data.map((information) => {
             if (information.location) {
-                return (
-                    <Marker
-                        key={information.id}
-                        title={`Information`}
-                        pinColor={getPinColorBySignal(information.signal)}
-                        coordinate={{
-                            latitude: information.location.x,
-                            longitude: information.location.y,
-                        }}
-                    >
-                        <Callout>
-                            <Text style={MapStyles.title}>Information</Text>
-                            <Text>Signal: {information.signal}</Text>
-                            <Text>Provider: {information.provider}</Text>
-                        </Callout>
-                    </Marker>
-                );
+                return getCircleBySignalStrength(information);
             }
         });
-        // add users position to list
-        dataMarker.push(
-            <Marker
-                title="your position"
-                key="own-data-marker"
-                pinColor="#4569ab"
-                coordinate={{
-                    latitude: this.props.location.x,
-                    longitude: this.props.location.y,
-                }}
-            />
-        );
 
+        // add users position to list
+        if (this.currentPositionHasChanged()) {
+            const position = (
+                <Marker
+                    title="your position"
+                    key="own-data-marker"
+                    coordinate={this.props.location}
+                />
+            );
+            this.setState({ currentPositionMarker: position });
```

Comment:
Should this be `this.setState({...this.state, currentPositionMarker: position });` ?

Reply:
Should be done by the ‘setState‘ method internally